h2o-core/src/main/java/water/parser/Parser.java

```diff
@@ -87,7 +87,10 @@ ParseWriter streamParseZip( final InputStream is, final StreamParseWriter dout,
     StreamParseWriter nextChunk = dout;
     int zidx = bvs.read(null,0,0); // Back-channel read of chunk index
     assert zidx==1;
-    while( is.available() > 0 ) {
+    int streamAvailable = is.available();
+    while(streamAvailable > 0) {
+      parseChunk(cidx++, din, nextChunk);
+      streamAvailable = is.available(); // Can (also!) rollover to the next input chunk
```

Comment:
In the old version:

If we already fully read the input stream (the ByteVec) and is.available() is called and returns 0 in the while-condition, the chunk will be marked as fully read. This won't be noticed by the code and we will keep writing to the same output chunk.

Simply put: anytime the ByteVec stream can rollover to the next chunk we need to check that and go to the next output chunk as well. Otherwise the output vector will have less chunks.